injection/sharedmain/main.go

```diff
@@ -129,7 +129,7 @@ func GetLeaderElectionConfig(ctx context.Context) (*leaderelection.Config, error
 
 // EnableInjectionOrDie enables Knative Injection and starts the informers.
 // Both Context and Config are optional.
-func EnableInjectionOrDie(ctx context.Context, cfg *rest.Config) context.Context {
+func EnableInjectionOrDie(ctx context.Context, cfg *rest.Config) (context.Context, []controller.Informer) {
```

Comment:
This is a breaking api change for all things that already use this and we can't do that for all the downstreams.

/hold

Reply:
I guess this logic only works if `StartInformers` is called immediately after this function. At least `sharedmain` does not do that, so it needs to be reverted. Callers like `chaosduck/main.go` however seems less a problem, though I still think it should be fixed.